back/app/controllers/web_api/v1/baskets_ideas_controller.rb

```diff
@@ -52,6 +52,51 @@ def update
     end
   end
 
+  def upsert
+    # 1. Create or get basket
+    idea = Idea.find(params[:idea_id])
+    participation_context = ParticipationContextService.new.get_participation_context(idea.project)
+    participation_context_type = participation_context.phase? ? 'Phase' : 'Project'
+
+    basket = Basket.find_or_initialize_by(
+      participation_context: participation_context,
+      participation_context_type: participation_context_type,
+      user: current_user
+    )
+    if basket.new_record?
+      if basket.save
+        SideFxBasketService.new.after_create basket, current_user
+      else
+        render json: { errors: baskets_idea.errors.details }, status: :unprocessable_entity
+        return
+      end
+    end
+
+    # 2. Create or update the baskets_idea
+    baskets_idea = BasketsIdea.find_or_initialize_by(
+      basket: basket,
+      idea: idea
+    )
+    record_is_new = baskets_idea.new_record?
+    authorize baskets_idea
+
+    baskets_idea.assign_attributes baskets_idea_params_for_update
```

Comment:
Should we remove the create and update endpoints? If so, I think the name `baskets_idea_params_for_update` is no longer relevant.

Reply:
Yes probably, but didn't want to just yet until Luuc confirms he can do everything with this endpoint